### ISSUE TITLE
Auto detect changes in csr.conf.template and recreate certs.

### ISSUE THREAD

**Issue by balchua** [ubuntu/microk8s#470]
This is a continuation of the modifiable csr PR. https://github.com/ubuntu/microk8s/pull/441
Related to issue https://github.com/ubuntu/microk8s/issues/421

**Comment by ktsakalozos** [comment]
Thank you @balchua

**Comment by ikari-pl** [comment]
In layman's terms, what do I do to make use of this feature? In one of the previous PRs I found manual steps in teh comments, that included the following:
```
    mcedit /var/snap/microk8s/671/certs/csr.conf.template
    rm /var/snap/microk8s/current/certs/csr.conf
    sudo snap set microk8s csr-refresh=true```
but I noticed that while the extra domain has been added to my cert, the extra IP has not. People are also mentioning a `external_ip.txt` file, that I do not have.

**Comment by balchua** [comment]
@ikari-pl all the user need to do is update the file `/var/snap/microk8s/current/certs/csr.conf.template` with your changes.  Once saved microk8s will automatically update the csr.conf and generate the certs accordingly.  After that it will restart the apiserver.

**Comment by ktsakalozos** [comment]
Hi @ikari-pl updating the template file should be enough to trigger the rendering of the csr.  As you can see in https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/actions/common/utils.sh#L170 IPs are extracted from the system's and places in the csr template replaceing the #MOREIPS tag. If you have your own set of IPs you can remove that tag from the csr template.

**Comment by ikari-pl** [comment]
Thanks for the super fast response. @ktsakalozos @balchua i guess it might be caused by the fact that, if I read the bash correctly, it overwrites IPs 3 and up, generating its own list starting with `IP.3` which is the number I have used :) I was able to work around this now.
Would it also work if I put `IP.anythingelse`?

**Comment by balchua** [comment]
@ikari-pl i think so, its just a way to make sure user dont override the previous key (IP or DNS) value. These DNS  and IP will be added to the certs.

**Comment by ikari-pl** [comment]
Yes, I just wanted to raise the point that just by editing the file I didn't know only numbers lower than 3 will remain, and 3 and up will get overwritten by the autogenerated sections. Maybe the auto ones could start with a high number, like 990? :)

**Comment by balchua** [comment]
Got what you meant.  From the `csr.conf.template` anything below the `#MOREIPS` are user supplied.  From there you can specify and key.
